Neos.Utility.ObjectHandling/Classes/TypeHandling.php

```diff
@@ -34,7 +34,7 @@ abstract class TypeHandling
     /**
      * @var array
      */
-    protected static $collectionTypes = ['array', \ArrayObject::class, \SplObjectStorage::class, Collection::class];
+    protected static $collectionTypes = ['array', \ArrayObject::class, \SplObjectStorage::class, Collection::class, \IteratorAggregate::class];
```

Comment:
Hm, would it maybe make sense to add `\Traversable` instead of `\IteratorAggregate`? What we try to decide with the collection types is if the value can be `foreach`'d through (right?), which `\Traversable` is providing. `\IteratorAggregate` only adds to this the possibility to externalize the iterator to achieve the traversal instead of implementing the iteration methods itself. Alternatively we should also **add** `\Iterator`, as that and `\IteratorAggregate` are the two primary implementation interfaces for `\Traversable` (https://www.php.net/manual/en/class.traversable.php).
We could btw then even remove the `\ArrayObject`, `\SplObjectStorage` and `Collection`, since they all implement either `\Iterator` or `\IteratorAggregate`, but keeping them is a slight optimization for those cases.
```suggestion
    protected static $collectionTypes = ['array', \ArrayObject::class, \SplObjectStorage::class, Collection::class, \Traversable::class];
```

Reply:
But `\Traversable` is an abstract class - it does not contain any method, so how can it be defined code wise (for ex `RoleIdentifiers`) that it can be iterated?

Reply:
It can't, it's an internal class, but all classes that implement `\Iterator` or ´\IteratorAggregate` are also `\Traversable`.
> This is an internal engine interface which cannot be implemented in PHP scripts. Either IteratorAggregate or Iterator must be used instead. When implementing an interface which extends Traversable, make sure to list IteratorAggregate or Iterator before its name in the implements clause.